src/shogun/distance/EuclideanDistance.cpp

```diff
@@ -177,3 +179,38 @@ float64_t CEuclideanDistance::distance_upper_bounded(int32_t idx_a, int32_t idx_
 
 	return result;
 }
+
+SGMatrix<float64_t> CEuclideanDistance::get_distance_matrix()
+{
+	REQUIRE(lhs->get_feature_class()==C_DENSE,
+		"Left hand side (was %s) has to be CDenseFeatures instance!\n", lhs->get_name());
+	REQUIRE(rhs->get_feature_class()==C_DENSE,
+		"Right hand side (was %s) has to be CDenseFeatures instance!\n", rhs->get_name());
+
+	REQUIRE(lhs->get_feature_type()==F_DREAL,
+		"Left hand side (was %s) has to be of double type!\n", lhs->get_name());
+	REQUIRE(rhs->get_feature_type()==F_DREAL,
+		"Right hand side (was %s) has to be double type!\n", rhs->get_name());
```

Comment:
Sorry I am bit late to comment on this but is it a good idea to put a hard REQUIRE in this method, when in the `init(...)` we only need the features to have DOT property active? Maybe have a fail-safe fallback mechanism for other features and add this Eigen3 specialization for dense features (interface should be the same for both). Also, this method should work for 32-bit float features also.